src/DynamoCoreWpf/Controls/StartPage.xaml.cs

```diff
@@ -270,6 +272,37 @@ internal void WalkDirectoryTree(System.IO.DirectoryInfo root, SampleFileEntry ro
             }
         }
 
+        /// <summary>
+        /// Sets the sampleDatasetsPath based on the value of sampleFolderPath
+        /// </summary>
+        private void SetSampleDatasetsPath()
+        {
+            try
+            {
+                var directoryInfo = new DirectoryInfo(sampleFolderPath);
+
+                // Traverse the directory tree upwards to locate the "samples" folder
+                while (directoryInfo != null && directoryInfo.Name != "samples")
+                {
+                    directoryInfo = directoryInfo.Parent;
+                }
+
+                if (directoryInfo != null && directoryInfo.Name == "samples")
+                {
+                    var datasetsPath = Path.Combine(directoryInfo.FullName, "Data");
+
+                    if (Directory.Exists(datasetsPath))
+                    {
+                        sampleDatasetsPath = datasetsPath;
+                    }
```

Comment:
So, we open a default page when dataset folder does not exist, or nothing happens?

Reply:
will falling back on the Desktop be sufficient? Any other suggestions?